pronotepy/dataClasses.py

```diff
@@ -761,7 +761,8 @@ def __init__(self, client: ClientBase, json_dict: dict) -> None:
         self.type: int = self._resolver(int, "G")  # 0 link, 1 file
 
         if self.type == 0:
-            self.url = self.name
+            url: str | None = self._resolver(str, "url", default=None)
+            self.url = self.name if url is None else url
```

Comment:
```suggestion
            url = self._resolver(str, "url", default=None)
            self.url: str = self.name if url is None else url
```
I'm not sure why the mypy check doesn't catch it, but the `|` syntax has been introduced only in python 3.10 (PEP 604). We're supporting python >=3.7, so we can't use it.

It's a local variable, so it's fine if it isn't typed. Linters can guess the type.